olp-cpp-sdk-dataservice-read/tests/StreamLayerClientImplTest.cpp

```diff
@@ -0,0 +1,251 @@
+/*
+ * Copyright (C) 2019 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+#include <gtest/gtest.h>
+
+#include <matchers/NetworkUrlMatchers.h>
+#include <mocks/CacheMock.h>
+#include <mocks/NetworkMock.h>
+#include <olp/core/client/OlpClientSettingsFactory.h>
+#include <olp/dataservice/read/CatalogClient.h>
+#include "StreamLayerClientImpl.h"
+
+namespace {
+using namespace olp::client;
+using namespace olp::dataservice::read;
+using namespace ::testing;
+using namespace olp::tests::common;
+
+constexpr auto kUrlLookupStream =
+    R"(https://api-lookup.data.api.platform.here.com/lookup/v1/resources/hrn:here:data::olp-here-test:hereos-internal-test-v2/apis/stream/v2)";
+
+constexpr auto kUrlStreamSubscribe =
+    R"(https://web.streaming.data.api.platform.here.com/stream/v2/catalogs/hrn:here:data::olp-here-test:hereos-internal-test-v2/layers/testlayer/subscribe?mode=serial)";
+
+constexpr auto kHttpResponseLookupStream =
+    R"jsonString([{"api":"stream","version":"v2","baseURL":"https://web.streaming.data.api.platform.here.com/stream/v2/catalogs/hrn:here:data::olp-here-test:hereos-internal-test-v2","parameters":{}}])jsonString";
```

Comment:
```suggestion
    R"jsonString([{"api":"stream","version":"v2","baseURL":"https://some.streaming.url/stream/v2/catalogs/hrn:here:data::olp-here-test:hereos-internal-test-v2","parameters":{}}])jsonString";
```
IMHO, I'm curious - why would you put real URL into tests? Same in other mocked responses.